mem_context.go

```diff
@@ -84,6 +87,9 @@ func NewMemUsageContext(
 			// number of obj props beyond which we should estimate mem usage
 			return objPropsLen > objPropsLenThreshold
 		},
+		ComputeSampleStep: func(totalItems int) int {
```

Comment:
[opt] I know this is the pattern we have, but we could just write this as a receiver method.

Reply:
Now that I'm thinking about this never mind, it's outside of scope for the PR